### PR TITLE
Implementing Documents endpoint

### PR DESCRIPTION
**Pull request by HarisaranG**
#12 
## PR Checklist

- [x] I have read and signed the [Contributor License Agreement](https://forms.gle/PZyiY5N2GDQU8GsV9).

### CODE REVIEW

File: lib/src/documents.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        bodyParameters: document, queryParams: ops);
+  }
+
+  Future<Map<String, dynamic>> update(Map<String, dynamic> document,
+      {Map<String, dynamic> options = const {}}) async {
+    Map<String, dynamic> ops = new Map<String, dynamic>.from(options)
+      ..addAll({"action": "update"});
+    return await _apicall.post(_endPoint(),
+        bodyParameters: document, queryParams: ops);
+  }
```
Comment (happy-san) [inline lib/src/documents.dart:36]: `update` function should call `ApiCall.patch` function instead according to the [docs](https://typesense.org/docs/0.20.0/api/documents.html#update-a-document)

**Review by happy-san — DISMISSED** [review]

File: lib/src/documents.dart (hunk 1 above)
Comment (happy-san) [reply lib/src/documents.dart:36]: Just checked the docs again. I got `Documents` confused with `Document`. All good!

**Comment by happy-san** [comment]
@HarisaranG I'll be merging your PR. I've made few changes to the ApiCall implementation, I'll handle the integration. As always, thank you for contributing!

P.S. Don't forget to checkmark the PR checklist. (I almost did it on your behalf, sorry 😅)

**Comment by HarisaranG** [comment]
I could go ahead and contribute to the remaining endpoints right ?

**Comment by happy-san** [comment]
Just wait for me to push my changes, I'll ping you then.

**Pull request merged**